src/view/com/composer/photos/EditImageDialog.web.tsx

```diff
@@ -0,0 +1,191 @@
+  const [aspect, setAspect] = React.useState(initialAspect)
+
+  const isEmpty = !crop || (crop.width || crop.height) === 0
+  const isNew = initialCrop ? true : !isEmpty
+
+  const onPressSubmit = React.useCallback(async () => {
+    const result = await manipulateImage(image, {
+      crop: !isEmpty
+        ? {
+            originX: (crop.x * source.width) / 100,
+            originY: (crop.y * source.height) / 100,
+            width: (crop.width * source.width) / 100,
+            height: (crop.height * source.height) / 100,
+          }
+        : undefined,
+    })
+
+    onChange(result)
+    control.close()
+  }, [crop, isEmpty, image, source, control, onChange])
+
+  const changeAspect = (next: number) => {
+    if (next !== aspect) {
+      setAspect(next)
+      setCrop(undefined)
+    }
+  }
+
+  return (
+    <Dialog.Inner label={_(msg`Edit image`)}>
+      <Dialog.Close />
+
+      <Text style={[a.text_2xl, a.font_bold, a.leading_tight, a.pb_sm]}>
+        <Trans>Edit image</Trans>
+      </Text>
+
+      <View style={[a.align_center]}>
+        <ReactCrop
+          aspect={aspect}
+          crop={crop}
+          onChange={(_, next) => setCrop(next)}
+          className="ReactCrop--no-animate">
+          <img src={source.path} style={{maxHeight: `50vh`}} />
+        </ReactCrop>
+      </View>
+
+      <View style={[a.mt_md, a.gap_md]}>
+        <View style={[a.flex_row]}>
+          <ToolbarButton
+            label={_(msg`Original`)}
+            onPress={() => changeAspect(sourceAspect)}>
+            <Svg width="24" height="24" viewBox="0 0 24 24" fill="none">
+              <Path
+                d="M3 5H21V19H3V5Z"
+                stroke={
+                  aspect === sourceAspect
+                    ? t.palette.primary_600
+                    : t.palette.white
+                }
+                strokeWidth="2"
+                strokeLinejoin="round"
+              />
+            </Svg>
+          </ToolbarButton>
+
+          <ToolbarButton
+            label={_(msg`Wide`)}
+            onPress={() => changeAspect(16 / 9)}>
+            <Svg width="24" height="24" viewBox="0 0 24 24" fill="none">
+              <Path
+                d="M1 7H23V17H1V7Z"
+                stroke={
+                  aspect === 16 / 9 ? t.palette.primary_600 : t.palette.white
+                }
+                strokeWidth="2"
+                strokeLinejoin="round"
+              />
+            </Svg>
+          </ToolbarButton>
+
+          <ToolbarButton
+            label={_(msg`Square`)}
+            onPress={() => changeAspect(1 / 1)}>
+            <Svg width="24" height="24" viewBox="0 0 24 24" fill="none">
+              <Path
+                d="M4 4H20V20H4V4Z"
+                stroke={
+                  aspect === 1 / 1 ? t.palette.primary_600 : t.palette.white
+                }
+                strokeWidth="2"
+                strokeLinejoin="round"
+              />
+            </Svg>
+          </ToolbarButton>
```

Comment:
not sure how the icons are being exported here so i just plopped it off central's react native template